python-scripts/src/main/python/dataproducts/util/utils.py

```diff
@@ -193,6 +193,73 @@ def get_tenant_info(result_loc_, org_search_, date_):
         print("Max retries reached...")
 
 
+def get_location_info(result_loc_, location_search_, date_):
+    """
+    get districts and state mapping of current environment
+    :param result_loc_: pathlib.Path object to store resultant CSV at
+    :param location_search_: host ip and port of server hosting location search API
+    :param date_: datetime object to pass to file path
+    :return: None
+    """
+    url = "{}api/data/v1/location/search".format(location_search_)
+    payload = """{
+        "request": {
+             "limit": 5000,
+             "filters": {
+                "type": ["district", "state"]
+             }
+        }
+    }"""
+    headers = {
+        'Accept': "application/json",
+        'Content-Type': "application/json",
+        'cache-control': "no-cache",
+        'Authorization': "Bearer {}".format(os.environ['API_KEY'])
+    }
+    retry_count = 0
+    while retry_count < 5:
+        retry_count += 1
+        try:
+            response = requests.request("POST", url, data=payload, headers=headers)
+            result = response.json()['result']['response']
+            states = pd.DataFrame(
+                    list(filter(lambda x: x['type'] == "state", result))
+                )
```

Comment:
Use either recursion of move the getLocation within the while loop into a method.